### PR TITLE
[pack] Fix admin download (3.x)

### PR DESCRIPTION
**Pull request by yojagad**
### Issue describing the changes in this PR

Part of fix for #6535

### Pull request checklist

* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

The reason this does not need a backport is because although the API doesn't work for both the versions, the reasons seems to be different. I'm in process of investigating why it doesn't work for 2.x

### CODE REVIEW

File: src/WebJobs.Script.WebHost/WebJobsApplicationBuilderExtension.cs [hunk 1]
```diff
@@ -41,7 +41,7 @@ public static IApplicationBuilder UseWebJobsScriptHost(this IApplicationBuilder
 
             // Specialization can change the CompatMode setting, so this must run later than
             // the PlaceholderSpecializationMiddleware
-            builder.UseWhen(context => httpBodyControlOptions.CurrentValue.AllowSynchronousIO, config =>
+            builder.UseWhen(context => httpBodyControlOptions.CurrentValue.AllowSynchronousIO || context.Request.IsAdminDownloadRequest(), config =>
```
Comment (ahmelsayed) [inline src/WebJobs.Script.WebHost/WebJobsApplicationBuilderExtension.cs:44]: I thought all of the code in the download api was using async methods all the way through. Or is it just the case that file IO in .net is always synchronous at some point?

**Review by ahmelsayed — APPROVED** [review]

File: src/WebJobs.Script.WebHost/WebJobsApplicationBuilderExtension.cs (hunk 1 above)
Comment (yojagad) [reply src/WebJobs.Script.WebHost/WebJobsApplicationBuilderExtension.cs:44]: Even if download APIs are async, IISServer seems to do a "Write" (and not WriteAsync) internally at some point to start writing to response stream which is causing an exception. This fails the downloadAPI silently and causes an empty zip to be downloaded.

Comment (ahmelsayed) [reply src/WebJobs.Script.WebHost/WebJobsApplicationBuilderExtension.cs:44]: Thanks for the details. I guess I only ever tested it without IIS in the picture, but that's good to know.

**Pull request merged**